sros2/sros2/_utilities.py

```diff
@@ -39,8 +39,11 @@ def create_symlink(*, src: pathlib.Path, dst: pathlib.Path):
     os.symlink(src, dst)
 
 
-def domain_id() -> str:
-    return os.getenv(_DOMAIN_ID_ENV, '0')
+def get_domain_id(domain_id: int = None) -> int:
+    validated_domain = domain_id
+    if (domain_id is None) or (domain_id < 0):
```

Comment:
if someone explicitly passed a negative domain ID, we should likely raise a ValueError to be explicit that this is not a valid value rather than silently replacing it with 0